cmd/admin-heal-ops.go

```diff
@@ -390,9 +388,66 @@ func newHealSequence(bucket, objPrefix, clientAddr string,
 		traverseAndHealDoneCh: make(chan error),
 		stopSignalCh:          make(chan struct{}),
 		ctx:                   ctx,
+		scannedItemsCount:     make(map[madmin.HealItemType]int64),
 	}
 }
 
+// resetHealStatusCounters - reset the healSequence status counters between
+// each monthly background heal scanning activity.
+// This is used only in case of Background healing scenario, where
+// we use a single long running healSequence which proactively heals
+// objects passed to the SourceCh.
+func (h *healSequence) resetHealStatusCounters() {
+	h.currentStatus.updateLock.RLock()
+	h.currentStatus.Items = []madmin.HealResultItem{}
+	h.currentStatus.updateLock.RUnlock()
+
+	h.lastSentResultIndex = 0
+	h.scannedItemsCount = make(map[madmin.HealItemType]int64)
+}
+
+// getScannedItemsCount - returns a count of all scanned items
+func (h *healSequence) getScannedItemsCount() int64 {
+	var count int64
+	for _, v := range h.scannedItemsCount {
+		count = count + v
+	}
+	return count
+}
+
+// getHealedItems - returns a map of count of successfully healed items
+// against item type
+func (h *healSequence) getHealedItems() map[madmin.HealItemType]int64 {
+	countMap := make(map[madmin.HealItemType]int64)
+	for _, hri := range h.currentStatus.Items {
+		// Error info is sent as String in Detail field.
+		// If Detail is "", we assume a successful healing
+		if hri.Detail == "" {
+			countMap[hri.Type]++
+		}
+	}
+	return countMap
+}
+
+// getHealFailedItems - returns a 2d map of count of items where healing
+// failed against drive state and drive endpoint
+func (h *healSequence) getHealFailedItems() map[string]map[string]int64 {
+	var countMap = make(map[string]map[string]int64)
+	countMap[madmin.DriveStateCorrupt] = make(map[string]int64)
+	countMap[madmin.DriveStateMissing] = make(map[string]int64)
+	countMap[madmin.DriveStateOffline] = make(map[string]int64)
+	for _, hri := range h.currentStatus.Items {
```

Comment:
h.currentStatus.Items is limited by memory, meaning it won't contain all data if the client (mc) won't consume regularly. I think the approach should be similar to scannedItemsCount where we update counters on the fly.